File: src/lib/entry/command/set-config-command.ts
import { CommandName } from "../../globals"
import JSONCommandType, { JSONCommandTypeArgs } from "./command-type/json-command-type"

const COMMAND_NAME_BYTE = new Uint8Array([CommandName.SET_CONFIG])

export default class SetConfigCommand extends JSONCommandType {
    constructor(args: JSONCommandTypeArgs) {
        if (!args.commandNameU8) {
            args.commandNameU8 = COMMAND_NAME_BYTE
        }
        super(args)
    }
}

File: external/logsrdcpp/src/entry/command/SetConfigCommand.cpp
#include "SetConfigCommand.h"

namespace logsrd {

SetConfigCommand::SetConfigCommand(JSONCommandTypeArgs args)
    : JSONCommandType([&]() {
        if (args.commandNameU8.empty()) {
            args.commandNameU8 = std::vector<uint8_t>{COMMAND_NAME_BYTE};
        }
        return args;
    }())
{}

} // namespace logsrd
